datafusion/expr-common/src/type_coercion/binary.rs

```diff
@@ -1449,6 +1455,22 @@ fn null_coercion(lhs_type: &DataType, rhs_type: &DataType) -> Option<DataType> {
     }
 }
 
+/// Resolves integer types to interval types for temporal arithmetic
+fn resolve_ints_to_intervals(
+    lhs: &DataType,
+    rhs: &DataType,
+) -> Option<(DataType, DataType, DataType)> {
+    use arrow::datatypes::DataType::*;
+    use arrow::datatypes::IntervalUnit::*;
+
+    match (lhs, rhs) {
+        // Handle integer + interval cases
+        (Int32 | Int64, _) => Some((Interval(DayTime), rhs.clone(), rhs.clone())),
```

Comment:
should this check rhs type to be something specific?

Reply:
I think so. I'll fix it